code/go/0chain.net/blobbercore/handler/file_command_add.go

```diff
@@ -36,7 +36,8 @@ func (cmd *AddFileCommand) IsAuthorized(ctx context.Context, req *http.Request,
 		return common.NewError("invalid_parameters",
 			"Invalid parameters. Error parsing the meta data for upload."+err.Error())
 	}
-	exisitingFileRef, _ := reference.GetReference(ctx, allocationObj.ID, fileChanger.Path)
+	// Update GetReference to GetReferenceID
+	exisitingFileRef, _ := reference.GetReferenceID(ctx, allocationObj.ID, fileChanger.Path)
```

Comment:
I think its better to atleast log error if any. In any case of err other than `gorm.ErrRecordNotFound` we should log actual error and probably return `common.NewError("some_code", "database error")

Reply:
@abouroumine 
I am not sure about using GetReferenceID. Have you checked it with `ProcessContent` function as well ?
In other `file_command*` code `cmd.existingFileRef` is used which infers that we might need to consider using `reference.GetReference`.

Above all, I think its best to have `single responsibility principle`. That's why `cmd.existingFileRef` should have been assigned before calling `IsAuthorize`.

Please make changes for `IsAuthorize` function to have single responsibility.

Reply:
I think unlike in file_command_update / file_command_delete, in file_command_add we are just checking if the File Ref exists. and we do not use it in ProcessContent. that is why GetReferenceID was used instead of GetReference.
Now that you have mentioned we can even reduce the columns selected for other file_command_* depending on the columns used for each command in each ProcessContent and other related functions.

Reply:
@abouroumine Its fine that you've called `GetReferenceID`. My point is that it is called inside `IsAuthorized` function making it hard to debug later on. We can call for existingRef just before `IsAuthorized` function is called and check for `ref` availability.